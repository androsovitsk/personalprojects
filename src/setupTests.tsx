import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import fetchMock from 'fetch-mock'
import TranslationProvider from './ui/components/TranslationProvider/TranslationProvider'
import LanguageTypes from './ui/components/TranslationProvider/types/LanguageTypes'
import resourceBundles from './locale'
import { createTheme, ThemeProvider } from '@mui/material'

beforeEach(() => {
  fetchMock.reset()
  jest.clearAllMocks()
})

const Wrapper: React.FC<React.PropsWithChildren> = ({ children }) => {
  return (
    <ThemeProvider theme={createTheme({})}>
      <TranslationProvider
        defaultLanguage={LanguageTypes.ENGLISH}
        resourceBundles={resourceBundles}
      >
        {children}
      </TranslationProvider>
    </ThemeProvider>
  )
}

export const renderWithProviders = (element: React.ReactElement) =>
  render(element, {
    wrapper: Wrapper
  })
