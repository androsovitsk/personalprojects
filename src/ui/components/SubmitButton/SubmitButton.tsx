import React from 'react'
import LoadingButton from '@mui/lab/LoadingButton'
import { FormikValues, useFormikContext } from 'formik'

const SubmitButton: React.FC<React.PropsWithChildren> = ({ children }) => {
  const { isSubmitting } = useFormikContext<FormikValues>()

  return (
    <LoadingButton type='submit' variant='contained' loading={isSubmitting}>
      {children}
    </LoadingButton>
  )
}

SubmitButton.displayName = 'SubmitButton'

export default SubmitButton
