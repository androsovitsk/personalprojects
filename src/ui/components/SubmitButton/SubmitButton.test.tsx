import { renderWithProviders } from '../../../setupTests'
import SubmitButton from './SubmitButton'
import { useFormikContext } from 'formik'

jest.mock('formik', () => ({
  ...jest.requireActual('formik'),
  useFormikContext: jest.fn()
}))

const useFormikContextMock = useFormikContext as jest.Mock
describe('SubmitButton', () => {
  it('should match the snapshot when the form is not submitting', () => {
    useFormikContextMock.mockReturnValue({ isSubmitting: false })

    const view = renderWithProviders(<SubmitButton />)
    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot when the form is submitting', () => {
    useFormikContextMock.mockReturnValue({ isSubmitting: true })

    const view = renderWithProviders(<SubmitButton />)
    expect(view.asFragment()).toMatchSnapshot()
  })
})
