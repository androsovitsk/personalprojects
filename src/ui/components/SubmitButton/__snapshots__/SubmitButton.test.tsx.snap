// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should match the snapshot when the form is not submitting 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    id=":r0:"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`SubmitButton should match the snapshot when the form is submitting 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiLoadingButton-root MuiLoadingButton-loading MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    disabled=""
    id=":r1:"
    tabindex="-1"
    type="submit"
  >
    <div
      class="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorCenter css-1yt7yx7-MuiLoadingButton-loadingIndicator"
    >
      <span
        aria-labelledby=":r1:"
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
        role="progressbar"
        style="width: 16px; height: 16px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </button>
</DocumentFragment>
`;
